development/ts-migration-dashboard/common/build-module-partitions.ts

```diff
@@ -208,76 +221,101 @@ function buildModulesWithLevels(
+      // Skip files that weren't on the original list of JavaScript files to
+      // convert, as we don't want them to show up on the status dashboard
+      if (
+        !npmPackageMatch &&
+        !allowedFilePaths.includes(
+          dependencyModuleId.replace(/\.tsx?$/u, '.js'),
+        )
+      ) {
+        continue;
+      }
 
-        if (existingChildModule === undefined) {
-          const childModule: Module = {
-            id: childModuleId,
-            dependents: [currentModule],
-            level: currentModule.level + 1,
-            isExternal: Boolean(npmPackageMatch),
-            hasBeenConverted: /\.tsx?$/u.test(childModuleId),
-          };
-          childModulesToFill.push(childModule);
-        } else {
-          if (currentModule.level + 1 > existingChildModule.level) {
-            existingChildModule.level = currentModule.level + 1;
-            // Update descendant modules' levels
-            childModulesToFill.push(existingChildModule);
-          } else {
-            // There is no point in descending into dependencies of this module
-            // if the new level of the module would be <= the existing level,
-            // because all of the dependencies from this point on are guaranteed
-            // to have a higher level and are therefore already at the right
-            // level.
-          }
+      let existingDependencyModule = modulesById[dependencyModuleId];
 
-          if (!existingChildModule.dependents.includes(currentModule)) {
-            existingChildModule.dependents.push(currentModule);
-          }
-        }
-      },
-    );
-    if (childModulesToFill.length > 0) {
-      modulesToFill.push(...childModulesToFill);
+      if (existingDependencyModule === undefined) {
+        existingDependencyModule = {
+          id: dependencyModuleId,
+          dependents: [currentModule],
+          dependencies: [],
+          level: currentModule.level + 1,
+          isExternal: Boolean(npmPackageMatch),
+          hasBeenConverted: /\.tsx?$/u.test(dependencyModuleId),
+        };
+        dependencyModulesToFill.push(existingDependencyModule);
+      } else if (currentModule.level + 1 > existingDependencyModule.level) {
+        existingDependencyModule.level = currentModule.level + 1;
+        // Update descendant modules' levels
+        dependencyModulesToFill.push(existingDependencyModule);
+      } else {
+        // There is no point in descending into dependencies of this module
+        // if the new level of the module would be <= the existing level,
+        // because all of the dependencies from this point on are guaranteed
+        // to have a higher level and are therefore already at the right
+        // level.
+      }
+
+      if (
+        !existingDependencyModule.dependents.some(
+          (m) => m.id === currentModule.id,
+        )
+      ) {
+        existingDependencyModule.dependents.push(currentModule);
+      }
+
+      if (
```

Comment:
The only part that's truly new in this file is this line, where, as we are iterating through the dependency of a given module, we make sure to associate that dependency with the module itself. This allows us to look up the dependencies later when we are rendering the visualization.